pipeline/sources/webcam_worker.py

```diff
@@ -0,0 +1,234 @@
+        # We add 4 hours to make sure that we really get the latest image. One is subtracted at the start of the loop,
+        # the remaining 3 are enough for any issue with UTC / local time / daylight-saving time / issues
+        # (which is max 2).
+        check_datetime = datetime.now(tz=ZoneInfo('Europe/Berlin')) + timedelta(hours=4)
+        while check_datetime > datetime.now(tz=ZoneInfo('Europe/Berlin')) - timedelta(days=self.config.keep_days):
+            # Jump one hour back at each iteration
+            check_datetime = check_datetime - timedelta(hours=1)
+
+            image_base_path = Path(
+                webcam_base_path,
+                f'{check_datetime.year:04d}',
+                f'{check_datetime.month:02d}',
+                f'{check_datetime.day:02d}',
+                f'{check_datetime.hour:02d}',
+            )
+            # If the base path does not exist, we can be sure that there are no images
+            if not image_base_path.exists():
+                continue
+
+            latest_image_path: Path | None = None
+            latest_image_datetime: datetime | None = None
+            for image_path in image_base_path.iterdir():
+                # We just want files
+                if not image_path.is_file():
+                    continue
+
+                image_name = image_path.name
+                # We just want images
+                if not image_name.endswith('.jpg'):
+                    continue
+
+                # We ignore empty files
+                if self.config.check_empty_files and image_path.stat().st_size == 0:
+                    continue
+                try:
+                    image_datetime = datetime.strptime(image_name[1:-4], '%y%m%d%H%M%S%f')
+                except ValueError:
+                    # If we have a value error, the file name is invalid, so we continue
+                    continue
+
+                if latest_image_datetime is None or image_datetime > latest_image_datetime:
+                    latest_image_datetime = image_datetime
+                    latest_image_path = image_path
+
+            # If we found an image: break, so we can set the symlink
+            if latest_image_path is not None and latest_image_datetime is not None:
+                return SymlinkItem(
+                    path=latest_image_path,
+                    name=webcam_base_path.name,
+                    filename=f'{webcam_base_path.name}.jpg',
+                    moment=latest_image_datetime,
+                )
+
+        # If we did not find an image, we return None
+        return None
+
+    def clean(self):
+        # Delete all old webcam images
+        self._run_command(
+            command=[
+                'find',
+                str(self.config.image_path),
+                '-mtime',
+                f'+{self.config.keep_days}',
+                '-type',
+                'f',
+                '-delete',
```

Comment:
This is super dangerous... What happens if someone accidently provides an image_path like '/'?

Reply:
Hmm. How to mitigate? Mounting the host / would be very very stupid, but, well, never underestimate people in this case, but: how to change?

Reply:
We could only delete files whose name and parent directory's name fit the scheme we expect.

Reply:
Ah, good idea. I added two things:

a) a check for files / directories which NEVER should be in the two dirs we handle
b) at the deletion function a match for the file name based on regex